bin/ch/Helpers.cpp

```diff
@@ -7,30 +7,41 @@
 #include <sys/stat.h>
 
 //TODO: x-plat definitions
-#ifdef _WIN32
 #define MAX_URI_LENGTH 512
+#ifdef _WIN32
+#define TTD_MAX_FILE_LENGTH MAX_PATH
 #define TTD_HOST_PATH_SEP "\\"
+#else
+#define TTD_MAX_FILE_LENGTH MAX_URI_LENGTH
+#define TTD_HOST_PATH_SEP "/"
+#endif
 
 void TTDHostBuildCurrentExeDirectory(char* path, size_t* pathLength, size_t bufferLength)
 {
-    wchar exePath[MAX_PATH];
-    GetModuleFileName(NULL, exePath, MAX_PATH);
+    char exePath[TTD_MAX_FILE_LENGTH];
+    PlatformAgnostic::SystemInfo::GetBinaryLocation(exePath, TTD_MAX_FILE_LENGTH);
 
-    size_t i = wcslen(exePath) - 1;
-    while(exePath[i] != _u('\\'))
+    size_t i = strlen(exePath) - 1;
+    while (exePath[i] != TTD_HOST_PATH_SEP[0] && i != 0)
     {
         --i;
     }
-
-    if(i * 3 > bufferLength)
+    if (i == 0)
     {
-        wprintf(_u("Don't overflow path buffer during conversion"));
+        wprintf(_u("Can't get current exe directory"));
         exit(1);
     }
-    *pathLength = utf8::EncodeInto((LPUTF8)path, exePath, (charcount_t)(i + 1));
+    if (i + 2 > bufferLength)
+    {
+        printf("Don't overflow path buffer during copy");
```

Comment:
`fprintf(stderr, ...` instead?

Reply:
Also printf is not allowed anymore in ChakraCore because it will cause an assertion on Windows because printf can't handle our upgrade to UTF-8 output.

Reply:
We prefer using `Output::Print`. At a minimum use `wprintf` and `fwprintf`

Reply:
I don't get the part.. why we use wprintf once the string is not unicode? i.e. the string above? All the `_u(x)` etc. stuff is an unnecessary hassle.

Reply:
It is probably okay because the string is ASCII, but as a general policy we don't use `printf`, and instead use `Output::Print` (which takes a unicode string).

Reply:
(I think in `ch` we don't have access to the `Output` class so we use `wprintf` instead.)

Reply:
So use `fwprintf(stderr, ...);`?

Reply:
Yeah, I think that would be best. @Penguinwizzard since you're making more changes in the area of stdlib output, what do you think?

Reply:
Yeah, use fwprintf(stderr, ...) for error for now; the _u is a hassle, but I'm not entirely sure if mixing printf and wprintf works nicely in all cases (especially since setting up output encoding properly is already a bit nuanced).

Reply:
I have changed both to `fwprintf(stderr, ...);`

Reply:
@Penguinwizzard Do you want me to change other occurrences of `fprintf` and `wprintf` in this file to `fwprintf`?

Reply:
@xiaoyinl up to you, but I'd say since it's out of scope of this change and it hasn't broken yet, if you want to make changes like that you could do it in a follow-up PR.

Reply:
I see.

Reply:
@xiaoyinl Thanks for offering, but I actually have a change in the works (but backlogged because of more important work) that does a lot of those changes.